postgres/datadog_checks/postgres/util.py

```diff
@@ -383,33 +383,43 @@ def get_list_chunks(lst, n):
 # Requires postgres 10+
 REPLICATION_STATS_METRICS = {
     'descriptors': [
-        ('application_name', 'wal_app_name'),
-        ('state', 'wal_state'),
-        ('sync_state', 'wal_sync_state'),
-        ('client_addr', 'wal_client_addr'),
+        ('replication.application_name', 'wal_app_name'),
+        ('replication.state', 'wal_state'),
+        ('replication.sync_state', 'wal_sync_state'),
+        ('replication.client_addr', 'wal_client_addr'),
+        ('replication_slot.slot_name', 'slot_name'),
+        ('replication_slot.slot_type', 'slot_type'),
     ],
     'metrics': {
-        'GREATEST (0, EXTRACT(epoch from write_lag)) as write_lag': (
-            'replication.wal_write_lag',
+        'GREATEST (0, EXTRACT(epoch from replication.write_lag)) as write_lag': (
+            'wal_write_lag',
             AgentCheck.gauge,
         ),
-        'GREATEST (0, EXTRACT(epoch from flush_lag)) AS flush_lag': (
-            'replication.wal_flush_lag',
+        'GREATEST (0, EXTRACT(epoch from replication.flush_lag)) AS flush_lag': (
+            'wal_flush_lag',
             AgentCheck.gauge,
         ),
-        'GREATEST (0, EXTRACT(epoch from replay_lag)) AS replay_lag': (
-            'replication.wal_replay_lag',
+        'GREATEST (0, EXTRACT(epoch from replication.replay_lag)) AS replay_lag': (
+            'wal_replay_lag',
             AgentCheck.gauge,
         ),
-        'GREATEST (0, age(backend_xmin)) as backend_xmin_age': (
-            'replication.backend_xmin_age',
+        'GREATEST (0, age(replication.backend_xmin)) as backend_xmin_age': (
+            'backend_xmin_age',
             AgentCheck.gauge,
         ),
     },
     'relation': False,
     'query': """
-SELECT application_name, state, sync_state, client_addr, {metrics_columns}
-FROM pg_stat_replication
+SELECT replication.application_name,
+replication.state,
+replication.sync_state,
+replication.client_addr,
+{metrics_columns},
```

Comment:
Order of columns is important here. You want the `metrics_columns` to be last as the first columns are used for the tags listed in `descriptors`.